src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -828,6 +829,80 @@ public static BinaryOperandsKind GetBinaryOperandsKind(BinaryOperationKind kind)
         }
     }
 
+    public interface IConversion
+    {
+        /// <summary>
+        /// Returns true if the conversion exists, either as an implicit or explicit conversion.
+        /// </summary>
+        bool Exists { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        bool IsIdentity { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit ('widening' in VB) conversion.
+        /// </summary>
+        bool IsImplicit { get; }
+
+        /// <summary>
+        /// True if and only if the conversion is an explicit ('narrowing' in VB) conversion.
+        /// </summary>
+        bool IsExplicit { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit numeric conversion or explicit numeric conversion. 
+        /// </summary>
+        bool IsNumeric { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit nullable conversion or explicit nullable conversion.
+        /// </summary>
+        bool IsNullable { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit reference conversion or explicit reference conversion.
+        /// </summary>
+        bool IsReference { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit user-defined conversion or explicit user-defined conversion.
+        /// </summary>
+        bool IsUserDefined { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit default ('null' in C#, 'nothing' in VB) literal conversion.
+        /// </summary>
+        bool IsDefault { get; }
+
+        /// <summary>
+        /// Returns the method that defines the user defined conversion, if any..
+        /// </summary>
+        IMethodSymbol MethodSymbol { get; }
+    }
+
+    public static class ConversionExtensions
```

Comment:
> Don't we already have types where we put compilation and semantic model extensions?

Not in this project we don't.   Well, we do have "ModelExtensions", however that lives in Microsoft.CodeAnalysis.  I can't add these extensions to that namespace as it would cause source level breaking changes as they would be seen at the same time as the extnesions we introduce there in the C# and VB compilation dlls.

I've worked around that by putting these into the Microsoft.CodeAnalysis.Semantics namespace.  This has two benefits:

1. No one is pulling this namespace in today, so the introduction of these extensions won't break any source.
2. You get this when you want to work with IConversion.  IConversion is in the "Microsoft.CodeAnalysis.Semantics" namespace, so it makes sense that tha'ts what you would import.

I can pull this into it's own file, but i didn't see a lot of value in it.  Here is where we define IConversion, so it seemed reasonable that hte extensions that would